src/Shared/Telemetry/TelemetryProcessor.ts

```diff
@@ -8,123 +8,124 @@ import { userContext } from "../../UserContext";
+      timestamp: timestamp,
+      data: JSON.stringify(data)
+    }
+  });
 
-  public static traceCancel(action: Action, data?: any, timestamp?: number): void {
-    sendMessage({
-      type: MessageTypes.TelemetryInfo,
-      data: {
-        action: Action[action],
-        actionModifier: ActionModifiers.Cancel,
-        timestamp: timestamp || Date.now(),
-        data: JSON.stringify(data)
-      }
-    });
-
-    appInsights.stopTrackEvent(Action[action], TelemetryProcessor.getData(data));
-  }
+  appInsights.startTrackEvent(Action[action]);
+  return timestamp;
+}
 
-  public static traceOpen(action: Action, data?: any, timestamp?: number): number {
-    const validTimestamp = timestamp || Date.now();
-    sendMessage({
-      type: MessageTypes.TelemetryInfo,
-      data: {
-        action: Action[action],
-        actionModifier: ActionModifiers.Open,
-        timestamp: validTimestamp,
-        data: JSON.stringify(data)
-      }
-    });
-
-    appInsights.startTrackEvent(Action[action]);
-    return validTimestamp;
-  }
+export function traceSuccess(action: Action, data?: unknown, timestamp?: number): void {
+  sendMessage({
+    type: MessageTypes.TelemetryInfo,
+    data: {
+      action: Action[action],
+      actionModifier: ActionModifiers.Success,
+      timestamp: timestamp || Date.now(),
+      data: JSON.stringify(data)
+    }
+  });
 
-  public static traceMark(action: Action, data?: any, timestamp?: number): number {
-    const validTimestamp = timestamp || Date.now();
-    sendMessage({
-      type: MessageTypes.TelemetryInfo,
-      data: {
-        action: Action[action],
-        actionModifier: ActionModifiers.Mark,
-        timestamp: validTimestamp,
-        data: JSON.stringify(data)
-      }
-    });
-
-    appInsights.startTrackEvent(Action[action]);
-    return validTimestamp;
-  }
+  appInsights.stopTrackEvent(Action[action], getData(data));
+}
 
-  private static getData(data: any = {}): any {
-    if (typeof data === "string") {
-      data = { message: data };
+export function traceFailure(action: Action, data?: unknown, timestamp?: number): void {
+  sendMessage({
+    type: MessageTypes.TelemetryInfo,
+    data: {
+      action: Action[action],
+      actionModifier: ActionModifiers.Failed,
+      timestamp: timestamp || Date.now(),
+      data: JSON.stringify(data)
     }
+  });
+
+  appInsights.stopTrackEvent(Action[action], getData(data));
+}
+
+export function traceCancel(action: Action, data?: unknown, timestamp?: number): void {
+  sendMessage({
+    type: MessageTypes.TelemetryInfo,
+    data: {
+      action: Action[action],
+      actionModifier: ActionModifiers.Cancel,
+      timestamp: timestamp || Date.now(),
+      data: JSON.stringify(data)
+    }
+  });
+
+  appInsights.stopTrackEvent(Action[action], getData(data));
+}
+
+export function traceOpen(action: Action, data?: unknown, timestamp?: number): number {
+  const validTimestamp = timestamp || Date.now();
```

Comment:
I was going to suggest nullish coalescing here `timestamp && Date.now()`, but a timestamp of 0 is probably not valid, so this is intentional?

Reply:
I think so. This was existing code, just formatting changes when the class got removed.